components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/ClaimUtil.java

```diff
@@ -305,6 +316,56 @@ private static String getSpMappedRoleClaim(ServiceProvider serviceProvider,
         return OIDCClaimUtil.getServiceProviderMappedUserRoles(serviceProvider, rolesList, claimSeparator);
     }
 
+    private static List<String> filterSharedRoles(ServiceProvider serviceProvider,
+                                                  List<String> userRoles) throws FrameworkException {
+
+        List<String> rolesAssociatedWithApp = new ArrayList<>();
+        if (CollectionUtils.isNotEmpty(userRoles)) {
```

Comment:
Let's refactor the code for better readbility.
Ex:
if (CollectionUtils.isEmpty(userRoles)) { 
   return rolesAssociatedWithApp;
}